Eng-574: trigger workflow GitHub integration

efiShtain opened the pull request.

efiShtain committed:
.

efiShtain committed:
.

efiShtain committed:
ENG-598: auto create default env for project (#138)

- Create a default env for project on project's creation
- Stop creating default env in manifest
- Remove double transalte error in transaction wrapper
- fix tests to match

efiShtain committed:
initial workflow api github

daabr commented on integrations/github/client.go:
```diff
@@ -246,5 +246,18 @@ func New(sec sdkservices.Secrets) sdkservices.Integration {
+		sdkmodule.ExportFunction(
+			"list_workflows",
+			i.listWorkflows,
+			sdkmodule.WithFuncDoc("https://docs.github.com/en/rest/actions/workflows#create-a-workflow-dispatch-event"),
```
```suggestion
			sdkmodule.WithFuncDoc("https://docs.github.com/en/rest/actions/workflows#list-repository-workflows"),
```

daabr commented on integrations/github/client.go:
```diff
@@ -246,5 +246,18 @@ func New(sec sdkservices.Secrets) sdkservices.Integration {
+			"list_workflows",
+			i.listWorkflows,
+			sdkmodule.WithFuncDoc("https://docs.github.com/en/rest/actions/workflows#create-a-workflow-dispatch-event"),
+			sdkmodule.WithArgs("owner", "repo"),
```
Can you also add "per_page" and "page"? (Both optional integers). I expect realistic usage of this will easily go over the default (1 page of 30 results).
Anyway, not as urgent as the core capability, so feel free to do it in a separate PR.

efiShtain replied:
done

daabr commented on integrations/github/workflows.go:
```diff
@@ -0,0 +1,67 @@
+	return sdktypes.Nothing, nil
+}
+
+func (i integration) listWorkflows(ctx context.Context, args []sdktypes.Value, kwargs map[string]sdktypes.Value) (sdktypes.Value, error) {
```
```suggestion
// https://docs.github.com/en/rest/actions/workflows#list-repository-workflows
func (i integration) listWorkflows(ctx context.Context, args []sdktypes.Value, kwargs map[string]sdktypes.Value) (sdktypes.Value, error) {
```

daabr reviewed (approved):
Approved, but see 2 tiny suggestions, and 1 request (not urgent for this specific PR).

itayd committed:
dispatcher options include env, not just envid (#132)

- dispatcher option receives env, not just env id. this allows for
integrations to direct events specifically to certain envs without
knowing their id, without using tokens.
- http integration use of ^.
- token is not mandated in manifest and cmd.
- fix some old StrictParseXID to Strict(ParseXID).

itayd committed:
http integration: return body functions (#137)

instead of returning `body_json`, `body_bytes` etc, which is... not
pretty, we return now functions that return the appropriate data. These
are functions and not just values since in the future we'll do it lazily
and not just duplicating the payload.

For that purpose, "const function" constructors are implemented that
just return their data as values.

itayd committed:
ENG-592: allow to specify http routes in triggers (#136)

- triggers have a new field: `data`. arbitrary data can be specified
there which the dispatcher can read.
- dispatcher knows to expect `data` for triggers of http integrations.
it would parse that and match the incoming http request against a
pattern.
- cripser definition of what passed to event hander: instead of just
`data`, now also `trigger` and `event. `data` is now the combined
`trigger.data` and `event.data`.

efiShtain committed:
Update integrations/github/workflows.go

Co-authored-by: Daniel Abraham <daniel@autokitteh.com>

efiShtain committed:
Update integrations/github/client.go

Co-authored-by: Daniel Abraham <daniel@autokitteh.com>

efiShtain committed:
add per_page and page to github list workflows integration
